### PR TITLE
S8 T2 V1: UI for the Availability Times of Support Staff

### PR DESCRIPTION
**Pull request by bradleyhallett**
Scrum Board Ticket: S8 T2

Story: Story 8 Available times

Description: Create UI that would consist of having a display of available days, where for the selected date, available times would be in half-hour time slots.

**I hard coded the availability times for the time being, planning to have the code access the back end with the worker-availability api in the future, ran out of time in this sprint.**

Testing Criteria: Tested code by running locally and checking the React App to make sure everything works properly.

How Reviewers can Test Code:

Run the code locally 

**Test code still needs to be implemented**

### CODE REVIEW

**Comment by Kyle199910** [comment]
Create appointment page runs fine

File: react-ui/src/components/CreateAppointmentForm.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+class CreateAppointmentForm extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        workerId: 8000000, // taken from Amy's test for the api
+        schoolId: 1,       //will need to implement a page before this to pass these values through
+        studentId: 0, // check with Melissa if already stored in props
+        workerTimeslotId : 0, 
+        purpose: "", // Max 300 => input size is 300
```
Comment (nancyprabhakar) [inline react-ui/src/components/CreateAppointmentForm.js:24]: Hi Bradley,

I recommend adding a message on the UI specifying the character limit. How would the user know if their input is cut off mid-message due to exceeding the 300 character limit? (Ideally, there would be a backwards character counter, but due to complexity that is out of scope.)

File: react-ui/src/components/CreateAppointmentForm.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        });
+      });
+  }
+  //add an else if statement for successful form submissiom but unsuccessful appointment submission (api backend)
+  //have the user redo the book appointment process
+  render() {
+    if (this.state.successfulAppointment && this.state.formSubmission)
+      return (
+        <div>
+          Success!
```
Comment (nancyprabhakar) [inline react-ui/src/components/CreateAppointmentForm.js:79]: I would recommend replacing the output message with 'Successful form submission!' to clarify that this doesn't mean a successful appointment submission.

File: react-ui/src/components/CreateAppointmentForm.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    axios.post(`/api/worker-availability?${queryString.stringify(params)}`)
+    .then(res => {
+    console.log(res.data);
+    let availableTimeArray = res.data;
+    this.setState({
+      availableTime: availableTimeArray
+    });
+  });
+}
```
Comment (nancyprabhakar) [inline react-ui/src/components/CreateAppointmentForm.js:163]: What is this section intended for?

File: react-ui/src/components/CreateAppointmentForm.js (hunk 1 above)
Comment (nancyprabhakar) [inline react-ui/src/components/CreateAppointmentForm.js:24]: Would the student be required to enter WorkerID in order to view the worker's available times? The student should be able to search by worker name, and would not know the worker's ID.

**Review by nancyprabhakar — CHANGES_REQUESTED** [review]
Hi Bradley, 
I've made a few comments; Please review!

File: react-ui/src/components/CreateAppointmentForm.js (hunk 1 above)
Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.js:24]: I will get to work on that!

Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.js:24]: No there will be a page before this where the student will select a worker they want to work with, then the WorkerID will be passed through.

File: react-ui/src/components/CreateAppointmentForm.js (hunk 3 above)
Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.js:163]: This is for future development, right now the availability times are hard coded. This was the beginning process of accessing the database to retrieve the available times for a given worker and school.

File: react-ui/src/components/CreateAppointmentForm.js (hunk 2 above)
Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.js:79]: The if statement will output success only if the appointment information was in fact sent to the data base to be stored and if the form was submitted. So the "success" message is for both the a successful form submission and appointment submission.

File: react-ui/src/components/CreateAppointmentForm.js (hunk 1 above)
Comment (bradleyhallett) [reply react-ui/src/components/CreateAppointmentForm.js:24]: @nancyprabhakar I have added a note for the character limit for the user! It is directly under the "reason" field.

Comment (nancyprabhakar) [reply react-ui/src/components/CreateAppointmentForm.js:24]: Hi Bradley, this is perfect. I agree that a a backwards character counter would be out of scope. Your workaround will provide the clarity the user needs to ensure they make an informed message. Thank you for making this change!

Comment (nancyprabhakar) [reply react-ui/src/components/CreateAppointmentForm.js:24]: The workerID values should stay private and not be surfaced to the student for data privacy reasons. This is a good flow to ask the student to select the worker name (which they will know via the the worker description) and pass the workerID internally.

File: react-ui/src/components/CreateAppointmentForm.js (hunk 2 above)
Comment (nancyprabhakar) [reply react-ui/src/components/CreateAppointmentForm.js:79]: I see, and what is the message for form submission and/or appointment submission failure?

File: react-ui/src/components/CreateAppointmentForm.js (hunk 3 above)
Comment (nancyprabhakar) [reply react-ui/src/components/CreateAppointmentForm.js:163]: This story was rather large, and should be divided into smaller sections. I will note that the scope of this PR  contains the following:
In scope: 
1. Create UI to display the worker availability
2. Worker availability data is derived from hard coded values

Out of scope:
1. The retrieval of worker availability from the database is excluded for sprint 1.

**Review by nancyprabhakar — APPROVED** [review]
